### DISCUSSION TITLE
Is there a clean example of desktop app navigation in MS Windows App SDK 1.0

### DISCUSSION THREAD

**Discussion by mstasak** [microsoft/WindowsAppSDK#2219] [Q&A]
If such exists, I'm batting 0.000 finding it.  Something which works on a desktop app, without shoehorning UWP classes into it, and builds under current VS2022 using documented references and usings.  Preferably with enough examples to cover menubars, breadcrumbs, navigate to-go back, open nonmodal page in new window, persisted vs reconstructed pages, and even modal popup windows.  My hunting gets me a mixed mess of UWP, Xamarin, WPF, WINUIx, etc. none of which seem to work in WINUI3/WINAPPSDK (what is RootFrame! ;/ ), so I think we need a clear document describing best practices, even if it just advocates 3rd party/roll-your-own approaches.

**Comment by castorix** [comment]
The [XAML Controls Gallery](https://docs.microsoft.com/en-us/samples/microsoft/xaml-controls-gallery/xaml-controls-gallery/) has many good examples (it is the UWP version of WinUI 3 Controls Gallery, but the code is the same (I even used it in VB.NET))

**Reply by mstasak** [reply]
Thank you castorix!  I did see several gallery apps and tried to build this one and got errors (I don't remember if they were build-time or run-time, regardless I downloaded the store version and studied it, and also tried to copy and paste fragments into an app with issues).  I didn't keep a history, but I think I ran into differences between UWP and WinAppSDK, and didn't try long enough to figure them out.  Also I was looking for basic pulldown menu behavior, not a Metro experience.  I thought (perhaps wrongly) that UWP APIs were outdated and should be avoided.  Since then, I've realized some fixes are as simple as changing windows.ui.xaml.something to microsoft.ui.xaml.something.  I still say the best way forward isn't clear enough, especially for a xaml-newbie like me.  Backing up, I feel some fundamental working docs are still needed.  And maybe some clear marking of to which UI framework(s) each class/example applies.

**Reply by castorix** [reply]
> Since then, I've realized some fixes are as simple as changing windows.ui.xaml.something to microsoft.ui.xaml.something. 

Yes, it is in the doc : [Migrate from UWP to the Windows App SDK](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/migrate-to-windows-app-sdk/migrate-to-windows-app-sdk-ovw)
I usually just do that and I can copy/paste blocks of XAML without problem

**Comment by FinnurHrafn** [comment]
I have found the samples in this blog very helpful in getting started with WinUI 3:
https://xamlbrewer.wordpress.com/
